src/domain/users/index.types.d.ts

```diff
@@ -64,6 +64,10 @@ type PhoneMetadataValidator = {
   validate(phoneMetadata: PhoneMetadata | null): true | ApplicationError
 }
 
+type IpMetadataValidator = {
+  validate(userIP: UserIPs): true | ApplicationError
```

Comment:
`ApplicationError` is the catch all error we return from the `app` use cases and should not be present in the `domain` layer. Domain layer primitives should return specific errors in their signature. I know its not consistent anywhere but since this is adding code lets stick to it.

Having domain return `ApplicationError` creates a circular dependency from app -> domain -> app where we only want it to go app -> domain. You could also fix this in the `PhoneMetadataValidator` in line 64 while you're at this code.

Reply:
Understood. I've changed both the `PhoneMetadataValidator` and `IpMetadataValidator` to use `ValidationError` instead of `ApplicationError`.